src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationReloadToken.cs

```diff
@@ -33,11 +33,60 @@ public class ConfigurationReloadToken : IChangeToken
         /// <param name="callback">The callback to invoke.</param>
         /// <param name="state">State to be passed into the callback.</param>
         /// <returns>The <see cref="CancellationToken"/> registration.</returns>
-        public IDisposable RegisterChangeCallback(Action<object?> callback, object? state) => _cts.Token.Register(callback, state);
+        public IDisposable RegisterChangeCallback(Action<object?> callback, object? state)
+        {
+#if NETCOREAPP || NETSTANDARD2_1
+            try
+            {
+                return _cts.Token.UnsafeRegister(callback, state);
```

Comment:
We should write tests for the new functionality (ensure that AsyncLocals aren't flown as expected).